consensus/istanbul/backend/announce.go

```diff
@@ -1161,10 +1180,59 @@ func (sb *Backend) generateEnodeCertificateMsg(version uint) (*istanbul.Message,
 	if err := msg.Sign(sb.Sign); err != nil {
 		return nil, err
 	}
-	logger.Trace("Generated Istanbul Enode Certificate message", "enodeCertificate", enodeCertificate, "address", msg.Address)
+
 	return msg, nil
 }
 
+// generateEnodeCertificateMsg generates an enode certificate message with the enode
+// this node is publicly accessible at. If this node is proxied, the proxy's
+// public enode is used.
+func (sb *Backend) generateEnodeCertificateMsgs(version uint) (map[enode.ID]*istanbul.Message, error) {
+	logger := sb.logger.New("func", "generateEnodeCertificateMsg")
+
+	externalEnodes := make(map[enode.ID]*enode.Node)
+	if sb.IsProxiedValidator() {
+		if valProxyAssignments, err := sb.proxyEngine.GetValidatorProxyAssignments(); err != nil {
```

Comment:
The refactor I mentioned before could also be applied to this call of `GetValidatorProxyAssignments`